src/workspacer/Config/ConfigHelper.cs

```diff
@@ -58,14 +60,52 @@ private static string LoadConfig()
             return file;
         }
 
+        private static Assembly _configurationAssembly = default;
+
+        private static IConfigurationBuilder LoadConfigurationAssembly()
+        {
+            IConfigurationBuilder builder = default;
+            var configurationAssemblyPath = GetPathInUserFolder(ConfigAssemblyName);
+            var exists = File.Exists(configurationAssemblyPath);
+            if (exists)
+            {
+                // handle double initialization
+                if (_configurationAssembly == default)
+                {
+                    // prevent locking issues when copying the configuration assembly while workspacer is running.
+                    var copiedAssembly = GetPathInUserFolder(ConfigLoadedAssemblyName);
+                    if (File.Exists(copiedAssembly))
+                    {
+                        File.Delete(copiedAssembly);
+                    }
+                    File.Copy(configurationAssemblyPath, copiedAssembly);
+                    configurationAssemblyPath = copiedAssembly;
+                    _configurationAssembly = Assembly.LoadFile(configurationAssemblyPath);
```

Comment:
From my understanding of things, `Assembly.LoadFile` actually caches the loaded assembly.

This would mean that compiling a new config and doing a workspacer reload wont actually load the new configuration.

Isn't that a bit counter-intuitive?

Reply:
ah yeah good point I should probably create a new AssemblyLoadContext, and dispose/create anew.
I don't tend to use reload tbh as I am usually running it with a debugger attached, and actively making code changes.

I was under the impression that while it is indeed cached it keeps a lock on the assembly file on disk, I noticed that it would remain locked when I was running in release mode, perhaps I should read the assembly into a memory stream and load it from there instead of doing the copy to disk

Reply:
> I don't tend to use reload tbh as I am usually running it with a debugger attached, and actively making code changes

In general, I think when making a config it's common to try different things interactively until you find what works. It would be nice to be able to support that flow.

> perhaps I should read the assembly into a memory stream and load it from there instead of doing the copy to disk

That's probably a simpler approach, yeah.